pkg/publisher/s3/publisher_test.go

```diff
@@ -24,13 +26,16 @@ import (
 )
 
 const bucket = "bacalhau-test-datasets"
-const prefix = "integration-tests-publisher/"
 const region = "eu-west-1"
 const endpoint = "https://s3-eu-west-1.amazonaws.com"
 
 var jobID = uuid.NewString()
 var executionID = uuid.NewString()
 
+// Ensure unique prefix
+var timestamp = time.Now().UTC().Format("20060102T150405") // yyyyMMddThhmmss
+var prefix = fmt.Sprintf("integration-tests-publisher-%s/%s/", timestamp, executionID)
+
```

Comment:
Looks great. The only concern I have is we might end up with a noisy bucket if some tests fail to delete their test directories. We are using the bucket for some static datasets used by other test scenarios.

My recommendation is to change the prefix to `fmt.Sprintf("integration-tests-publisher/%s-%s/", timestamp, executionID)`

Which will give a better console experience and limit the noise to one directory/prefix. An alternative is using a dedicate bucket just for this test scenario, but that is on overkill, and changing the prefix will give us the same benefit.